### PR TITLE
Cleanup webrequestpscmdlet.common.cs

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary

Small cleanup for better readibility

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

**Comment by CarloToso** [comment]
Should I remove redundant brackets ? (example: `if (PassThru && (OutFile == null))`)

**Comment by iSazonov** [comment]
@CarloToso You could look CodeFactor issues reported for the file and fix some ones.

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 1]
```diff
@@ -237,21 +237,14 @@ public abstract partial class WebRequestPSCmdlet : PSCmdlet
...
-        private int _maximumRedirection = -1;
+        public virtual int MaximumRedirection { get; set; } = -1;
 
         /// <summary>
         /// Gets or sets the MaximumRetryCount property, which determines the number of retries of a failed web request.
         /// </summary>
         [Parameter]
         [ValidateRange(0, int.MaxValue)]
-        public virtual int MaximumRetryCount { get; set; }
+        public virtual int MaximumRetryCount { get; set; } = 0;
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:247]: It is default. Please revert.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 2]
```diff
@@ -410,97 +389,84 @@ public virtual string CustomMethod
         internal virtual void ValidateParameters()
         {
             // sessions
-            if ((WebSession != null) && (SessionVariable != null))
+            if (WebSession != null && SessionVariable != null)
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:392]: We could use `is` and `is not` patterns in the file.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 3]
```diff
@@ -762,14 +703,17 @@ private Uri PrepareUri(Uri uri)
 
         private static Uri CheckProtocol(Uri uri)
         {
-            if (uri == null) { throw new ArgumentNullException(nameof(uri)); }
+            if (uri == null) 
+            { 
+                throw new ArgumentNullException(nameof(uri)); 
+            }
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:709]: Not for the PR (we could update all code base) - we could use new `ArgumentNullException.ThrowIfNull` and such others.

**Review by iSazonov — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs (hunk 1 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: Done

**Comment by CarloToso** [comment]
Squashed

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 4]
```diff
@@ -389,89 +389,89 @@ public abstract partial class WebRequestPSCmdlet : PSCmdlet
...
             // Authentication
-            if (UseDefaultCredentials && Authentication != WebAuthenticationType.None)
+            if (UseDefaultCredentials && Authentication is not WebAuthenticationType.None)
             {
                 ErrorRecord error = GetValidationError(WebCmdletStrings.AuthenticationConflict, "WebCmdletAuthenticationConflictException");
                 ThrowTerminatingError(error);
             }
 
-            if (Authentication != WebAuthenticationType.None && Token != null && Credential != null)
+            if (Authentication is not WebAuthenticationType.None && Token is not null && Credential is not null)
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:405]: Please revert patterns like `Authentication is not WebAuthenticationType.None`.

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:405]: Or better move this in follow PR after we merge that.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: Done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 5]
```diff
@@ -585,17 +545,15 @@ internal virtual void PrepareSession()
...
                 // save the session back to the PS environment if requested
                 PSVariableIntrinsics vi = SessionState.PSVariable;
                 vi.Set(SessionVariable, WebSession);
             }
 
-            //
             // handle credentials
-            //
-            if (Credential != null && Authentication == WebAuthenticationType.None)
+            if (Credential is not null && Authentication is WebAuthenticationType.None)
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:556]:
```suggestion
            if (Credential is not null && Authentication == WebAuthenticationType.None)
```

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 6]
```diff
@@ -1751,7 +1710,7 @@ internal long SetRequestContent(HttpRequestMessage request, XmlNode xmlNode)
         {
             ArgumentNullException.ThrowIfNull(request);
 
-            if (xmlNode == null)
+            if (xmlNode is null)
                 return 0;
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:1714]:
```suggestion
            if (xmlNode is null)
            {
                return 0;
            }
```

**Review by iSazonov — APPROVED** [review]

**Pull request merged**